src/ui/camera.js

```diff
@@ -758,10 +785,24 @@ class Camera extends Evented {
 
         tr.bearing = bearing;
 
-        const p0world = tr.project(LngLat.convert(p0));
-        const p1world = tr.project(LngLat.convert(p1));
+        const coord0 = LngLat.convert(p0);
+        const coord1 = LngLat.convert(p1);
+        const coord2 = new LngLat(coord0.lng, coord1.lat);
+        const coord3 = new LngLat(coord1.lng, coord0.lat);
+
+        const p0world = tr.project(coord0);
+        const p1world = tr.project(coord1);
+
+        const z0 = this.queryTerrainElevation(coord0);
+        const z1 = this.queryTerrainElevation(coord1);
+        const z2 = this.queryTerrainElevation(coord2);
+        const z3 = this.queryTerrainElevation(coord3);
+
+        const worldCoords = [
+            [p0world.x, p0world.y, Math.min(z0 || 0, z1 || 0, z2 || 0, z3 || 0)],
+            [p1world.x, p1world.y, Math.max(z0 || 0, z1 || 0, z2 || 0, z3 || 0)]
+        ];
```

Comment:
Looks like you're adding terrain support here, can you make a note of this in the PR?

Reply:
Updated, this is intended to be equivalent to the previous code path and not strictly new support.